src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -145,22 +171,74 @@ public boolean matchCacheableRequest(
         final DefaultCacheEntry cacheEntry = cachedEntries.get(requestHash);
         return satisfiedByCache(requestHeaders) &&
                 cacheEntry != null &&
-                cacheEntry.canServeRequest(requestHeaders, authScope);
-
+                cacheEntry.canServeRequest(requestHeaders, authScope) &&
+                !cacheEntry.isValid();
     }
 
 
     public void purge(
         int requestHash)
     {
-        DefaultCacheEntry cacheEntry = this.cachedEntries.remove(requestHash);
-        entryCount.accept(-1);
+        DefaultCacheEntry cacheEntry = cachedEntries.remove(requestHash);
         if (cacheEntry != null)
         {
+            final int collectionHash = cacheEntry.getCollectionHash();
+            Int2ObjectHashMap collection = links.get(collectionHash);
```

Comment:
In general, this naming is too specific. All we want is to ignore the query parameters, it doesn't have to imply anything about a `collection`.

Reply:
Done.